### ISSUE TITLE
Wrong argument type in TypeScript declaration?

### ISSUE THREAD

**Issue by draperunner** [JMPerez/promise-throttle#79]
Instead of taking a `Promise` like so:
```
add<T>(promise: Promise<T>, opts?: QueueOptions): Promise<T>
```

isn't it correct that `add` should take a _function that returns_ a `Promise<T>`?
```
add<T>(promiseCreator: () => Promise<T>, opts?: QueueOptions): Promise<T>
```

@IlyaSemenov

**Comment by draperunner** [comment]
Would you like a PR on this, @JMPerez ?

**Comment by JMPerez** [comment]
Absolutely, @draperunner!

**Comment by JMPerez** [comment]
Fixed in version 1.1.2 of the library.
